examples/basic_timelock_ibe/backend/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+}
+
+#[update(guard = is_authenticated)]
+fn place_bid(lot_id: u128, encrypted_amount: Vec<u8>) -> Result<(), String> {
+    let bidder = ic_cdk::caller();
+    let now = ic_cdk::api::time();
+
+    LOTS.with_borrow(|lots| match lots.get(&lot_id) {
+        Some(LotInformation {
+            status: LotStatus::Open,
+            creator,
+            end_time,
+            ..
+        }) if creator != bidder && now < end_time => Ok(()),
+        Some(LotInformation { creator, .. }) if creator == bidder => {
+            Err("lot creator cannot bid".to_string())
+        }
+        Some(_) => Err("lot is closed".to_string()),
+        None => Err("lot not found".to_string()),
+    })?;
+
+    if encrypted_amount.len() > 1000 {
+        return Err("encrypted amount is too large to be valid".to_string());
+    }
+
+    BIDS_ON_LOTS.with_borrow_mut(|bids| {
+        bids.insert(
+            (lot_id, bidder),
+            Bid::Encrypted(EncryptedBid {
+                encrypted_amount,
+                bidder,
+            }),
+        );
+    });
+
+    Ok(())
+}
+
+#[update]
+fn start_with_interval_secs(secs: u64) {
+    let secs = std::time::Duration::from_secs(secs);
+    ic_cdk_timers::set_timer_interval(secs, || ic_cdk::spawn(close_one_lot_if_any_is_open()));
+}
+
+async fn close_one_lot_if_any_is_open() {
+    let root_ibe_public_key =
+        if let Some(key) = VETKD_ROOT_IBE_PUBLIC_KEY.with_borrow(|key| key.clone()) {
+            key
+        } else {
+            get_root_ibe_public_key().await
+        }
+        .into_vec();
+
+    let now = ic_cdk::api::time();
+    let lot_to_close: Option<LotId> = OPEN_LOTS_DEADLINES.with_borrow_mut(|open_lots_deadlines| {
+        open_lots_deadlines
+            .iter()
+            .take_while(|(deadline, _)| *deadline <= now)
+            .next()
+            .map(|(deadline, lot_to_close)| {
+                // remove the lot from the open lots deadlines to prevent double processing
+                open_lots_deadlines.remove(&deadline);
+                lot_to_close
+            })
+    });
+
+    if let Some(lot_id) = lot_to_close {
+        let encrypted_bids: Vec<EncryptedBid> = BIDS_ON_LOTS.with_borrow(|bids| {
+            bids.range((lot_id, Principal::management_canister())..)
+                .take_while(|((this_lot_id, _), _)| *this_lot_id == lot_id)
+                .map(|(_, bid)| match bid {
+                    Bid::Encrypted(encrypted_bid) => encrypted_bid,
+                    Bid::Decrypted(_) => panic!("bug: decrypted bid in a closed lot"),
+                })
+                .collect()
+        });
+
+        let decrypted_bids = decrypt_bids(lot_id, encrypted_bids, root_ibe_public_key).await;
+
+        let status = match decrypted_bids
+            .iter()
+            .rev() // reverse the bids to get the *oldest* maximum bid
+            .max_by(|x, y| x.amount.cmp(&y.amount))
+        {
+            Some(winner_bid) => LotStatus::ClosedWithWinner(winner_bid.bidder),
```

Comment:
As discussed offline, this doesn't actually return the _oldest_ maximum bid.

Thanks for the fix in https://github.com/dfinity/vetkd-devkit/pull/89.